lighthouse-core/gather/driver/prepare.js

```diff
@@ -0,0 +1,177 @@
+ * @param {LH.Gatherer.FRProtocolSession} session
+ * @return {Promise<void>}
+ */
+async function dismissJavaScriptDialogs(session) {
+  session.on('Page.javascriptDialogOpening', data => {
+    log.warn('Driver', `${data.type} dialog opened by the page automatically suppressed.`);
+
+    session
+      .sendCommand('Page.handleJavaScriptDialog', {
+        accept: true,
+        promptText: 'Lighthouse prompt response',
+      })
+      .catch(err => log.warn('Driver', err));
+  });
+
+  await session.sendCommand('Page.enable');
+}
+
+/**
+ * @param {LH.Gatherer.FRProtocolSession} session
+ * @param {{url: string}} navigation
+ * @return {Promise<{warnings: Array<LH.IcuMessage>}>}
+ */
+async function resetStorageForNavigation(session, navigation) {
+  /** @type {Array<LH.IcuMessage>} */
+  const warnings = [];
+
+  // Reset the storage and warn if there appears to be other important data.
+  const warning = await storage.getImportantStorageWarning(session, navigation.url);
+  if (warning) warnings.push(warning);
+  await storage.clearDataForOrigin(session, navigation.url);
+  await storage.clearBrowserCaches(session);
+
+  return {warnings};
+}
+
+/**
+ * Prepares a target for a particular navigation by resetting storage and setting throttling.
+ *
+ * This method assumes `prepareTargetForNavigationMode` has already been invoked.
+ *
+ * @param {LH.Gatherer.FRProtocolSession} session
+ * @param {LH.Config.Settings} settings
+ * @param {Pick<LH.Config.NavigationDefn, 'disableThrottling'|'disableStorageReset'|'blockedUrlPatterns'> & {url: string}} navigation
+ */
+async function prepareNetworkForNavigation(session, settings, navigation) {
+  const status = {msg: 'Preparing network conditions', id: `lh:gather:prepareNetworkForNavigation`};
+  log.time(status);
+
+  if (navigation.disableThrottling) await emulation.clearThrottling(session);
+  else await emulation.throttle(session, settings);
+
+  // Set request blocking before any network activity.
+  // No "clearing" is done at the end of the navigation since Network.setBlockedURLs([]) will unset all if
+  // neccessary at the beginning of the next navigation.
+  const blockedUrls = (navigation.blockedUrlPatterns || []).concat(
+    settings.blockedUrlPatterns || []
+  );
+  await session.sendCommand('Network.setBlockedURLs', {urls: blockedUrls});
+
+  const headers = settings.extraHeaders;
+  if (headers) await session.sendCommand('Network.setExtraHTTPHeaders', {headers});
+
+  log.timeEnd(status);
+}
+
+/**
+ * Prepares a target to be analyzed in navigation mode by enabling protocol domains, emulation, and new document
+ * handlers for global APIs or error handling.
+ *
+ * This method should be used in combination with `prepareTargetForIndividualNavigation` before a specific navigation occurs.
+ *
+ * @param {LH.Gatherer.FRTransitionalDriver} driver
+ * @param {LH.Config.Settings} settings
+ */
+async function prepareTargetForNavigationMode(driver, settings) {
```

Comment:
the same as `setupDriver` but with storage reset extracted to `prepareTargetForIndividualNavigation`